src/ngraph/op/fused/squeeze.cpp

```diff
@@ -38,25 +38,35 @@ void op::Squeeze::pre_validate_and_infer_types()
     auto data = input_value(0);
     auto axes_node = input_value(1).get_node_shared_ptr();
 
-    if (data.get_partial_shape().is_dynamic() || !axes_node->is_constant())
+    bool dynamic_data_rank = data.get_partial_shape().rank().is_dynamic();
+    bool dynamic_data_shape = data.get_partial_shape().is_dynamic();
+
+    bool axes_is_constant = axes_node->is_constant();
+    bool axes_is_empty_constant =
+        (axes_is_constant) ? as_type_ptr<op::Constant>(axes_node)->cast_vector<int64_t>().empty()
+                           : false;
+
+    if (dynamic_data_rank || !axes_is_constant || (dynamic_data_shape && axes_is_empty_constant))
     {
         set_output_type(0, get_input_element_type(0), PartialShape::dynamic());
         return;
     }
 
-    auto data_shape = data.get_shape();
+    uint64_t data_rank = data.get_partial_shape().rank().get_length();
+    auto data_partial_shape = data.get_partial_shape();
 
     // Get value of axes from Constant
     auto axes_constant = as_type_ptr<op::Constant>(axes_node);
```

Comment:
This line would be up above with my suggested refactor.

Reply:
Done